resolver/dns/dns_resolver.go

```diff
@@ -338,9 +341,8 @@ func chosenByPercentage(a *int) bool {
 	if a == nil {
 		return true
 	}
-	s := rand.NewSource(time.Now().UnixNano())
-	r := rand.New(s)
-	return r.Intn(100)+1 <= *a
+	randomGen.Seed(time.Now().UnixNano())
```

Comment:
I don't think this is necessary.  The source has already been seeded, and you don't want to re-seed the RNG every time it's used.